### DISCUSSION TITLE
Welcome to a-shell Discussions!

### DISCUSSION THREAD

**Discussion by holzschu** [holzschu/a-shell#533] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by cdhdds** [comment]
Is there a way to use aliases in a-shell?  Sorry if obvious, I have searched - no luck.  I created .bash_profile with aliases but no luck.  Love the app!

**Comment by holzschu** [comment]
You have the `alias` command, so you can create aliases on the go: `alias nano=pico`. 

You can also place alias commands in the `.bashrc` file. The content of that file will be executed (line by line) for each new window.

**Comment by cdhdds** [comment]
Thanks, great app, moving some python files from macosx.  Neat to have them on iPad Pro

**Comment by cdhdds** [comment]
Hi Nicolas,
The alias command works fine, but I have been unable get aliases to work with a .bashrc file.  On default there was no .bashrc file so I created one and put aliases there but they are not read.  I quit the window several times with no luck.  Maybe the .bashrc file is somewhere that I haven’t found.
Thanks ~ Dudley

**Comment by holzschu** [comment]
I checked, and it works. The `.bashrc` file has to be in the starting directory (the one where you arrive with `cd`). Just sending the app to the background (pressing the home button) and bringing it forward (clicking on the app) is not enough to reload the `.bashrc` file, you need to either create a new window (`newWindow`, only on iPads) or quit the app (iPads: `exit` or click on the top of the screen, then select "close", iPhones: press home button twice, then slide a-Shell upwards).

**Comment by cdhdds** [comment]
Ok found my problem. I has spaces in the alias. My bad. Thanks for your
help.

**Comment by holzschu** [comment]
You can have spaces if you use the "=" syntax. I use `alias ll=ls -l` for example (but `alias ls l=ls -l` will not work).

**Comment by RAlex81** [comment]
How to delete history?

**Reply by holzschu** [reply]
On an iPhone, "exit" does wipe the window content and history.
On an iPad, you have to go in Settings and uncheck the box "Keep screen content between sessions", then exit the window.

**Comment by Abo5** [comment]
We want ruby programming language install in a new version of app

```
[Documents]$ ruby
ruby: command not found
[Documents]$ 
```

**Comment by MasComida** [comment]
Hi Nicolas,
Thank you for a great app. Is it possible to have this keyboard feature like in iSh app to use one hand typing?
![IMG_0013](https://github.com/holzschu/a-shell/assets/121099425/caae3d20-3ee4-4220-ab87-1fdbeb9e4da5)

**Reply by holzschu** [reply]
I'm not sure what this feature is exactly. If you could provide more information?

**Comment by MasComida** [comment]
Hi,
Thank you for replying. Its basically when I minimize the keyboard to use one hand keyboard typing would still like to see the A-shell tab, ctrl, esc, arrow keys and etc at the bottom like how iSh app does it.
Thank you for your time 🙏
Sent from my iPhone

On May 30, 2024, at 5:28 PM, Nicolas Holzschuch ***@***.***> wrote:

﻿

I'm not sure what this feature is exactly. If you could provide more information?

**Reply by holzschu** [reply]
Thanks. Now I understand your problem. I did a bit of experimenting. a-Shell has two different toolbars, and you can switch between them using the Settings app. The default on all recent iPads is the "System" toolbar, which is managed by iPadOS and looks better with external keyboards. If you deactivate this settings, you get another toolbar, which looks almost the same with the on-screen keyboard. When you switch to the floating keyboard, iPadOS deactivates the "System" toolbar, so you don't see it (because this toolbar is associated with the keyboard). But if you're using the other toolbar, it will stay on screen (because we manage it).

![image](https://github.com/holzschu/a-shell/assets/8070736/7807cbe2-cbfd-4f52-a2b2-6851803238bc)

Short version: open the Settings app, deactivate the "System" toolbar, switch to floating keyboard and enjoy one-hand-typing. You can also configure the toolbar to have more buttons on screen, using the "config" command. 

![image](https://github.com/holzschu/a-shell/assets/8070736/f2ee8e89-b966-4c47-a47d-dd3785a9ff20)

If you really like the "System" toolbar, there's another option: take out your stylus, start writing on screen. The keyboard disappears, the toolbar stays on screen. Type on the icon towards the left of the toolbar which has you keyboar name (En/Us, or Es/Es... you get the idea). On the menu that pops up, chose "Show keyboard". The iPad now shows a system toolbar and a floating keyboard (and you can now put down your stylus... or use it to write on screen, a-Shell works with iPadOS writing recognition). 

![image](https://github.com/holzschu/a-shell/assets/8070736/afdc094b-271e-40ae-a68d-6f4698b2e52b)

**Comment by MasComida** [comment]
It worked! I'm incredibly grateful for your help. Wishing you abundant success in all your endeavors! 🙏

Sent from my iPhone

On May 31, 2024, at 12:31 AM, Nicolas Holzschuch ***@***.***> wrote:

﻿

Thanks. Now I understand your problem. I did a bit of experimenting. a-Shell has two different toolbars, and you can switch between them using the Settings app. The default on all recent iPads is the "System" toolbar, which is managed by iPadOS and looks better with external keyboards. If you deactivate this settings, you get another toolbar, which looks almost the same with the on-screen keyboard. When you switch to the floating keyboard, iPadOS deactivates the "System" toolbar, so you don't see it (because this toolbar is associated with the keyboard). But if you're using the other toolbar, it will stay on screen (because we manage it).

Short version: open the Settings app, deactivate the "System" toolbar, switch to floating keyboard and enjoy one-hand-typing. You can also configure the toolbar to have more buttons on screen, using the "config" command.

If you really like the "System" toolbar, there's another option: take out your stylus, start writing on screen. The keyboard disappears, the toolbar stays on screen. Type on the icon towards the left of the toolbar which has you keyboar name (En/Us, or Es/Es... you get the idea). On the menu that pops up, chose "Show keyboard". The iPad now shows a system toolbar and a floating keyboard (and you can now put down your stylus... or use it to write on screen, a-Shell works with iPadOS writing recognition).

**Comment by ghost** [comment]
Is it possible to get into iPhone's(no jailbreak) root directory using a-shell?

**Reply by holzschu** [reply]
The short answer is: yes, but you won't get very far. 
The long answer: just type `cd /` and you are in the iPhone root directory. You can explore a few of the sub-directories, but you will quickly encounter forbidden directories ("Operation not permitted"). 

Possibly the most interesting one is `/System/Library` which contains a lot of frameworks and information, including all system fonts in `/System/Library/Fonts`.
